lib/index.js

```diff
@@ -22,11 +22,24 @@ var AppsFlyer = module.exports = integration('AppsFlyer')
  */
 
 AppsFlyer.ensure(function(msg, settings) {
-  if (contains(msg.library(), 'ios') && !settings.appleAppID) {
+
+  if (!msg.device().type) {
+    return this.reject('Events to AppsFlyer must specify a device type as a contextual property');
+  }
+
+  if (msg.device().type.toLowerCase() !== 'ios' && msg.device().type !== 'android') {
+    return this.reject('AppsFlyer integration is currently only supported for ios or android devices');
+  }
+
+  if (!msg.device().advertisingId) {
+    return this.invalid('Events to AppsFlyer must specify an advertisingId as a contextual property');
+  }
+
+  if (msg.device().type.toLowerCase() ==='ios' && !settings.appleAppID) {
```

Comment:
might be nice to just make a helper function that takes in the device object and returns a boolean value to make this code a bit more readable and cleaner